plugins/processors/aws/ec2/ec2.go

```diff
@@ -124,6 +124,20 @@ func (r *AwsEc2Processor) Init() error {
 		return errors.New("no tags specified in configuration")
 	}
 
+	for _, tag := range r.ImdsTags {
```

Comment:
It looks like the changes to this file are needed in this branch but wouldn't be a problem in current master.

Reply:
That's right. I believe now we need to do these tag checks in Init rather than connect, this was the error we were getting with the previous setup: 

This was the error we were getting previously: 
```
021/07/21 20:52:58 D! [] Initializing AWS EC2 Processor
2021/07/21 20:53:03 D! [processors.aws_ec2] Initializing AWS EC2 Processor
--- FAIL: TestLoadingConfig (4.27s)
    ec2_test.go:69: 
        	Error Trace:	ec2_test.go:69
        	Error:      	Received unexpected error:
        	            	error parsing aws_ec2, failed getting instance identity document: operation error ec2imds: GetInstanceIdentityDocument, exceeded maximum number of attempts, 3, request send failed, Get "http://169.254.169.254/latest/dynamic/instance-identity/document": dial tcp 169.254.169.254:80: connect: host is down
        	Test:       	TestLoadingConfig
FAIL
FAIL	github.com/influxdata/telegraf/plugins/processors/aws/ec2	9.737s
```

Reply:
Went ahead and moved these changes to a separate pr so we can get it merged earlier, hope that is alright: https://github.com/influxdata/telegraf/pull/9576

Reply:
Thats awesome, thank you!